plotters/src/style/colors/colormaps.rs

```diff
@@ -0,0 +1,252 @@
+                    // Next calculate a normalized value between 0.0 and 1.0
+                    let t = (h - min)/(max-min);
+                    let approximate_index = t * (FloatType::from_usize(self.colors.len()).unwrap() - FloatType::one()).max(FloatType::zero());
+                    // Calculate which index are the two most nearest of the supplied value
+                    let index_lower = approximate_index.floor().to_usize().unwrap();
+                    let index_upper = approximate_index.ceil().to_usize().unwrap();
+                    // Calculate the relative difference, ie. is the actual value more towards the color of index_upper or index_lower?
+                    let relative_difference = approximate_index.ceil() - approximate_index;
+                    // Interpolate the final color linearly
+                    calculate_new_color_value!(relative_difference, self.colors, index_upper, index_lower, $color_type)
+                }
+            }
+        )+
+    }
+}
+
+implement_color_scale_for_derived_ColorMap!{RGBAColor, RGBColor, HSLColor}
+
+
+macro_rules! count {
+    () => (0usize);
+    ($x:tt $($xs:tt)* ) => (1usize + count!($($xs)*));
+}
+
+
+macro_rules! define_colors_from_list_of_values_or_directly{
+    ($color_type:tt, $(($($color_value:expr),+)),+) => {
+        [$($color_type($($color_value),+)),+]
+    };
+    ($($color_complete:tt),+) => {
+        [$($color_complete),+]
+    };
+}
+
+
+macro_rules! implement_linear_interpolation_color_map{
+    ($color_scale_name:ident, $color_type:tt) => {
+        impl<FloatType: std::fmt::Debug + Float + FromPrimitive + ToPrimitive> ColorMap<$color_type, FloatType> for $color_scale_name {
+            fn get_color_normalized(&self, h: FloatType, min: FloatType, max: FloatType) -> $color_type {
+                // Ensure that we do have a value in bounds
+                let h = h.max(min).min(max);
+                // Make sure that we really have a minimal value which is smaller than the maximal value
+                assert_eq!(min<max, true);
+                // Next calculate a normalized value between 0.0 and 1.0
+                let t = (h - min)/(max-min);
+                let approximate_index = t * (FloatType::from_usize(Self::COLORS.len()).unwrap() - FloatType::one()).max(FloatType::zero());
+                // Calculate which index are the two most nearest of the supplied value
+                let index_lower = approximate_index.floor().to_usize().unwrap();
+                let index_upper = approximate_index.ceil().to_usize().unwrap();
+                // Calculate the relative difference, ie. is the actual value more towards the color of index_upper or index_lower?
+                let relative_difference = approximate_index.ceil() - approximate_index;
+                // Interpolate the final color linearly
+                calculate_new_color_value!(relative_difference, Self::COLORS, index_upper, index_lower, $color_type)
+            }
```

Comment:
There's a lot of duplicated code shared with `implement_color_scale_for_derived_ColorMap` that should probably be moved into a function.

Reply:
I have encapsulated the functionality into a function just as you described.